src/main/resources/org/openjdk/jextract/impl/resources/RuntimeHelper.java.template

```diff
@@ -2,81 +2,38 @@
 
 import java.lang.foreign.Linker;
 import java.lang.foreign.FunctionDescriptor;
-import java.lang.foreign.GroupLayout;
 import java.lang.foreign.SymbolLookup;
 import java.lang.foreign.MemoryLayout;
 import java.lang.foreign.MemorySegment;
-import java.lang.foreign.Arena;
-import java.lang.foreign.SegmentAllocator;
 import java.lang.foreign.ValueLayout;
 import java.lang.invoke.MethodHandle;
 import java.lang.invoke.MethodHandles;
 import java.lang.invoke.MethodType;
-import java.io.File;
-import java.nio.file.Path;
-import java.nio.charset.StandardCharsets;
-import java.util.Arrays;
-import java.util.Optional;
-import java.util.stream.Stream;
 
 import java.lang.foreign.AddressLayout;
 import java.lang.foreign.MemoryLayout;
 
-import static java.lang.foreign.Linker.*;
 import static java.lang.foreign.ValueLayout.*;
 import static java.lang.Long.MAX_VALUE;
 
 final class RuntimeHelper {
 
-    private static final Linker LINKER = Linker.nativeLinker();
-    private static final ClassLoader LOADER = RuntimeHelper.class.getClassLoader();
     private static final MethodHandles.Lookup MH_LOOKUP = MethodHandles.lookup();
     private static final SymbolLookup SYMBOL_LOOKUP;
-    private static final SegmentAllocator THROWING_ALLOCATOR = (x, y) -> { throw new AssertionError("should not reach here"); };
     static final AddressLayout POINTER = ValueLayout.ADDRESS.withTargetLayout(MemoryLayout.sequenceLayout(MAX_VALUE, JAVA_BYTE));
 
-    final static SegmentAllocator CONSTANT_ALLOCATOR =
-            (size, align) -> Arena.ofAuto().allocate(size, align);
-
     static {
-        #LOAD_LIBRARIES#
         SymbolLookup loaderLookup = SymbolLookup.loaderLookup();
-        SYMBOL_LOOKUP = name -> loaderLookup.find(name).or(() -> LINKER.defaultLookup().find(name));
+        Linker linker = Linker.nativeLinker();
```

Comment:
Question: with RuntimeHelper being so small now, and given one of the stated goal is to allow for better composition of different extractions, would it not be convenient to move all these definitions in the header class and ditch `RuntimeHelper` completely? If we do that, we might also avoid generating the varargs definitions unless we need them, so that the header class would only include minimal additional code.

Reply:
I thought about this, and I think the main question is: in which header class do we generate the helpers if we have multiple classes? I wasn't sure if we'd be happy with duplicating the helpers in every header class (users might complain about the duplication). But, maybe the majority of cases will only ever see a single header class, so there's still a benefit.

I'll take another look at that. The conditional code generation is also an interesting benefit.

Reply:
For fields that is not possible though, since we have to put those at the start of the header class to avoid illegal forward references, but we don't know when we create the first header class whether we are going to need multiple header classes. When we have only 1 header class, the first header is the 'root' into which the helpers should go, when we have multiple headers, the second header class is the root.

I think maybe it's simpler to just generate the helper methods and fields like `SYMBOL_LOOKUP` once per header class. Or use static methods with embedded Holder classes for the fields. We could also have a special last header that everything else inherits, i.e. just have the last header extend RuntimeHelper essentially. But when I got to that point in my thought process, I thought: why not just keep RuntimeHelper then.